content/tutorials/flutter-cryptocurrency-app.textile

```diff
@@ -0,0 +1,720 @@
+
+h3. Subscribe to Ably Channels
+
+Let’s break down the previous function and explain more how we are subscribing to the cryptocurrency channel in 4 steps:
+
+1. Get the list of realtime channels for your account, then get a specific one by its name
+
+```[flutter]
+ably.RealtimeChannel channel = _realtime.channels.get('[product:ably-coindesk/crypto-pricing]$coinCode:usd');
+```
+
+In a real App it’s probably a good idea to check if we got the channel that we requested.
+
+2. Subscribe to the chosen channel
+
+```[flutter]
+final Stream<ably.Message> messageStream = channel.subscribe();
+```
+
+3. The returned type from @subscribe()@ is a @Stream<Message>@, which feels a bit odd because we just subscribed to something. In reality the @subscribe@ tells Ably to start transmitting data.
+
+4. So we register a listener for this message Stream, and use `where` to filter null values.
+
+```[flutter]
+messageStream.where((event) => event.data != null).listen((message) {});
+```
+
+As we never stop listening to the channels in this app we can ignore the @StreamSubscription@ that is returned from @listen()@.
+
+Inside the listener, whenever a new message arrives, we call `updateCoin()` and pass it a new Coin mapped from the Message data.
+
+The return type of this function is a @List<CoinUpdates>@ which has the same length as @_coinTypes@ and with a @CoinUpdates@ object for every currency defined in @coinTypes@. 
+
+To be safe in case that this function is called more than once we wrap the @for@ loop in an @if@ that checks if we already have made our channel subscription.
+
+Now we have finished setting up all we need for the service, it’s time to see how we will use it to show the prices graphs. 
+
+To be able to access our services easily we use the service locator package @get_it@. You could also use @package/provider@ or any other solution that you are comfortable with.
+
+The following diagram visualizes how the data will flow from Ably to our App’s UI.
+
+h3(#register-get-it). Registering Services with get_it
+
+First step is to register the service using get_it:
+
+```[flutter]
+GetIt getIt = GetIt.instance;
+
+void main() {
+  getIt.registerSingletonAsync<AblyService>(() => AblyService.init());
+  runApp(MyApp());
+}
+```
+
+We will do that asynchronously in the main method, as we want this service to be available as soon as the app is launched.
+
+For more information on how asynchronous registration with @get_it@ works see https://github.com/fluttercommunity/get_it#asynchronous-singletons.
+
+As this is an asynchronous registration, it won’t be available to our UI immediately, so we will wait for it at some place before using it. For this we will use a @FutureBuilder@ which will show a loading spinner until get_it reports that all services are ready.
+
+```[flutter]
+FutureBuilder(
```

Comment:
show the whole widget code here so that they can copy it to a new file and add a dummy widget `GraphList` for now